src/skin.cpp

```diff
@@ -200,28 +124,25 @@ void SkinInfillAreaComputation::generateSkinAndInfillAreas()
  */
 void SkinInfillAreaComputation::generateSkinAndInfillAreas(SliceLayerPart& part)
 {
-    Polygons original_outline = (wall_line_count < 1) ? part.outline : part.outline.offset(-(wall_line_width_0 + (wall_line_count - 1) * wall_line_width_x));
-
-    // make a copy of the outline which we later intersect and union with the resized skins to ensure the resized skin isn't too large or removed completely.
-    Polygons upskin;
+    //Make a copy of the outline which we later intersect and union with the resized skins to ensure the resized skin isn't too large or removed completely.
+    Polygons top_skin;
     if (top_layer_count > 0)
     {
-        upskin = Polygons(original_outline);
+        top_skin = Polygons(part.inner_area);
     }
-    Polygons downskin;
+    Polygons bottom_skin;
     if (bottom_layer_count > 0 || layer_nr < LayerIndex(initial_bottom_layer_count))
     {
-        downskin = Polygons(original_outline);
+        bottom_skin = Polygons(part.inner_area);
```

Comment:
Why not move this logic in the `calculateBottomSkin` function and changed the function signature to `Polygons calculateBottomSkin(const SliceLayerPart& part)` so you can make it `const` and do:
`const Polygons bottom_skin = calculateBottomSkin(part);`

Reply:
This change would cause a copy of the polygon data. See above (and add comments to the one above if you disagree).